src/lv_draw/lv_img_decoder.c

```diff
@@ -705,7 +726,20 @@ static lv_res_t lv_img_decoder_built_in_line_indexed(lv_img_decoder_dsc_t * dsc,
     lv_color_t * cbuf = (lv_color_t *)buf;
     for(i = 0; i < len; i++) {
         val_act = (data_tmp[byte_act] & (mask << pos)) >> pos;
-        cbuf[i] = user_data->palette[val_act];
+
+        lv_color_t color = user_data->palette[val_act];
+#if LV_COLOR_DEPTH == 8 || LV_COLOR_DEPTH == 1
+        buf[i * LV_IMG_PX_SIZE_ALPHA_BYTE] = color.full;
+#elif LV_COLOR_DEPTH == 16
+        /*Because of Alpha byte 16 bit color can start on odd address which can cause crash*/
+        buf[i * LV_IMG_PX_SIZE_ALPHA_BYTE] = color.full & 0xFF;
+        buf[i * LV_IMG_PX_SIZE_ALPHA_BYTE + 1] = (color.full >> 8) & 0xFF;
+#elif LV_COLOR_DEPTH == 32
+        *((uint32_t *)&buf[i * LV_IMG_PX_SIZE_ALPHA_BYTE]) = color.full;
```

Comment:
It's really the same in the drawing routines written by me. I've chosen this ugly way to avoid having `buf8`, `buf16` and `buf32` for the 3 color formats.